src/device-manager/cocoa/NLWdmClient.mm

```diff
@@ -225,6 +227,48 @@ static void handleWdmClientError(
     [client dispatchAsyncDefaultFailureBlock:error];
 }
 
+static void onWdmClientFlushUpdateComplete(
+    void * appState, void * appReqState, uint16_t pathCount, nl::Weave::DeviceManager::WdmClientFlushUpdateStatus * statusResults)
+{
+    WDM_LOG_DEBUG(@"onWdmClientFlushUpdateComplete");
+
+    NLWdmClient * client = (__bridge NLWdmClient *) appReqState;
+
+    NSMutableArray * statusResultsList = [NSMutableArray arrayWithCapacity:pathCount];
+    NLGenericTraitUpdatableDataSink * dataSink = nil;
+    WDM_LOG_DEBUG(@"Failed path Count = %u\n", pathCount);
+
+    for (uint32_t i = 0; i < pathCount; i++) {
+        dataSink = [client getDataSink:(long long) statusResults[i].mpDataSink];
+        if (dataSink == nil) {
+            WDM_LOG_DEBUG(@"trait %d does not exist", i);
```

Comment:
Yes, you understanding is correct. 

just update the PR, I would return nil to upper application layer since this is callback function and i cannot directly return the error and don't wanna overide the error code from lower layer, then application can check this dataSink and raise the exception finally. here I adjust the log message in unexpected, trait %d does not exist in traitMap